src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs

```diff
@@ -110,8 +164,26 @@ public void RegisterLlvmCallbacks(IntPtr corInfoPtr, string outputFileName, stri
                 &getMangledMethodName,
                 &getSymbolMangledName,
                 &addCodeReloc,
-                &isRuntimeImport
-                );
+                &isRuntimeImport,
+                &getDocumentFileName,
+                &firstSequencePointLineNumber,
+                &getOffsetLineNumber
+            );
+        }
+
+        public void InitialiseDebugInfo(MethodDesc method, MethodIL methodIL)
+        {
+            MethodIL uninstantiatiedMethodIL = methodIL.GetMethodILDefinition();
+            if (methodIL != uninstantiatiedMethodIL)
```

Comment:
I think `methodCodeNodeNeedingCode.Method` should be the right instantiation and none of this should be needed here. The right place to fix the instantiation is at the point where the dependencies are added to the graph, not here. Mismatch between `methodCodeNodeNeedingCode.Method` and the actual method instantiation that the code is compiled for is just going to cause more problems later.

Reply:
Thanks, removed.